DigitalLearningSolutions.Data/DataServices/CompetencyLearningResourcesDataService.cs

```diff
@@ -31,12 +31,12 @@ public IEnumerable<int> GetCompetencyIdsByLearningResourceReferenceId(int learni
                 @"SELECT
                         CompetencyID
                     FROM CompetencyLearningResources
-                    WHERE LearningResourceReferenceID = @learningResourceReferenceId",
+                    WHERE LearningResourceReferenceID = @learningResourceReferenceId AND RemovedDate IS NULL",

```

Comment:
This method is used when adding resources to the action plan. Normally, we add a new item to LearningLogItems for each competency within the self assessment that has that resource listed. This change means that only competencies that have active competency learning resources will be linked to.

Not 100% sure if we need this change, but it made sense to me. Any thoughts/confirmation @SteveJacksonSoft  ?

Reply:
Sorry, I got confused on the table names when writing the above comment. I meant to say:

- We only add a single record to LearningLogItems, regardless of the number of linked competencies
- We create a new record in LearningLogItemCompetencies for each linked competency. 

So if a user adds an item to their action plan, only one copy of that resource will appear in the Current activities page. We have logic in place that should prevent a user from adding the item multiple times (except in cases where the resource has been completed/removed). The multiple records in LearningLogItemCompetencies are for supposedly for future features in a future signposting phase that I am not 100% sure of the details on (I think it might be for displaying tags of each linked competency on the resource cards).

Reply:
Ah cool. That sounds right, since a removed CompetencyLearningResource means that the resource is no longer linked to that competency. I would propose a name change to make it more conceptual, something like `GetCompetencyIdsLinkedToResource`.